include/picongpu/particles/atomicPhysics/ionizationPotentialDepression/StewartPyattIPD.hpp

```diff
@@ -227,7 +227,9 @@ namespace picongpu::particles::atomicPhysics::ionizationPotentialDepression
 
             // (eV * sim.unit.length()) / (eV * sim.unit.length()), not weighted
             // unitless, not weighted
-            float_X const K = constFactor / (temperatureTimesk_Boltzman * debyeLength);
+            float_X const K = (debyeLength <= 0._X || temperatureTimesk_Boltzman <= 0._X)
```

Comment:
Is this enough, or do you need an epsilon check as below? 
```suggestion
            float_X const K = ( (debyeLength * temperatureTimesk_Boltzman) < std::numeric_limits<float_X>::epsilon() )
```

Reply:
Depends.

The equality sign in there has virtually no impact in case the quantities are actually the result of floating point operations because hitting an exact 0 is unlikely. This could even mean that an analytically exact 0 acquires a tiny, positive contribution that would lift it over that limit. So, the original version more or less means `<` numerically (unless those values are set 0.0 explicitly because 0.0 is representable as a floating point number). If numerically computed 0.0 are a likely case, an epsilon check might be necessary.

Your version is inequivalent to the one above: If both quantities are large and negative, the original version will pass while your version will fail. Not sure if that's the logic error you wanted to point out.